docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst

```diff
@@ -1453,87 +1450,49 @@ Below, we detail the rules governing target template selection, then explain how
 transformations are performed, and finally detail the rules of metadata
 re-computation.
 
-Target Templates
-~~~~~~~~~~~~~~~~
-
-In a top-level exercise triggered by a Ledger API command, the target
-template is determined by the rules of package preference detailed in the
-:ref:`dynamic package
-resolution<dynamic-package-resolution-in-command-submission>` section of the
-smart contract upgrade documentation.
+Static Target Template Selection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 In a non-interface fetch or exercise triggered by the body of a choice, the
 target template is determined by the dependencies of the package that defines
 the choice. In other words, it is statically known.
 
-In an interface fetch or exercise, the target template is determined by the
-rules of package preference detailed in the :ref:`dynamic package
-resolution<dynamic-package-resolution-in-command-submission>` section of the
-smart contract upgrade documentation. In other words, it is dynamically
-selected.
-
-.. _example-1-1:
+Interface fetches and exercises on the other hand are subject to dynamic target
+template selection, as detailed in :ref:`the next
+section<dynamic-target-template-selection>`. However, operations acting on
+interface *values* — as opposed to IDs — are static. Their mode of operation is
+detailed below.
+
+Daml contracts are represented by one of two sorts of values at runtime:
+template values or interface values.
+
+* Template values are those whose concrete template type is statically
+  known. They are obtained by directly constructing a template record, or by a
+  call to ``fetch``. Their runtime representation is a simple record.
+* Interface values are those whose concrete template type is not fully 
+  statically known, aside from the fact that it implements a given interface.
+  They are obtained by applying ``toInterface`` to a template value.
```

Comment:
should we not say that this pair is not accessible by user ? I mean it should be consider as a blob.

Reply:
Good point, added this sentence:

Note that the type of interface values is opaque: while it is useful to
conceptualize interface values as pairs for defining the runtime semantics of
the language, their actual implementation may vary and is not exposed to the
user.